error: mingw32: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown

```
PS C:\vc\vcpkg> ./vcpkg install libpng
Your feedback is important to improve Vcpkg! Please take 3 minutes to complete our survey by running: vcpkg contact --survey
Computing installation plan...
The following packages will be built and installed:
    libpng[core]:x86-windows
Detecting compiler hash for triplet x86-windows...
Starting package 1/1: libpng:x86-windows
Building package libpng[core]:x86-windows...
Could not locate cached archive: C:\Users\Olaf\AppData\Local\vcpkg\archives\88\88af68ff88614130095eb55b4c9668b844331f7b.zip
-- Using cached C:/vc/vcpkg/downloads/glennrp-libpng-v1.6.37.tar.gz
-- Cleaning sources at C:/vc/vcpkg/buildtrees/libpng/src/v1.6.37-4612f3523c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vc/vcpkg/downloads/glennrp-libpng-v1.6.37.tar.gz
-- Applying patch use_abort.patch
-- Applying patch cmake.patch
-- Applying patch pkgconfig.patch
-- Applying patch pkgconfig.2.patch
-- Using source at C:/vc/vcpkg/buildtrees/libpng/src/v1.6.37-4612f3523c.clean
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Fixing pkgconfig
-- Acquiring MSYS Packages...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: C:/vc/vcpkg/downloads/tools/msys2/msys64/usr/bin/bash.exe --noprofile --norc -c "pacman -S --noconfirm --needed pkg-config"
    Working Directory: C:/vc/vcpkg/downloads/tools/msys2
    Error code: 1
    See logs for more information:
      C:\vc\vcpkg\buildtrees\libpng\msys-pacman-x86-windows-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_acquire_msys.cmake:153 (vcpkg_execute_required_process)
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:281 (vcpkg_acquire_msys)
  ports/libpng/portfile.cmake:117 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:79 (include)


Error: Building package libpng:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libpng:x86-windows
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.
```
```
PS C:\vc\vcpkg> git pull
Already up-to-date.
```
```
// C:\vc\vcpkg\buildtrees\libpng\msys-pacman-x86-windows-err.log
error: mingw32: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown
error: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" could not be looked up remotely
error: mingw64: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown
error: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" could not be looked up remotely
error: msys: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown
error: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" could not be looked up remotely
error: database 'mingw32' is not valid (invalid or corrupted database (PGP signature))
error: database 'mingw64' is not valid (invalid or corrupted database (PGP signature))
error: database 'msys' is not valid (invalid or corrupted database (PGP signature))

```

A clean vcpkg clone works fine, it'd be nice if vcpkg could recover from this failure itself though.

--- a/scripts/cmake/vcpkg_acquire_msys.cmake
+++ b/scripts/cmake/vcpkg_acquire_msys.cmake
@@ -36,7 +36,8 @@
 ## * [libvpx](https://github.com/Microsoft/vcpkg/blob/master/ports/libvpx/portfile.cmake)
 
 function(vcpkg_acquire_msys PATH_TO_ROOT_OUT)
-  set(TOOLPATH ${DOWNLOADS}/tools/msys2)
+  set(TIMESTAMP 20200812)
+  set(TOOLPATH ${DOWNLOADS}/tools/msys2-${TIMESTAMP})
   cmake_parse_arguments(_am "" "" "PACKAGES" ${ARGN})
 
   if(NOT CMAKE_HOST_WIN32)
@@ -145,7 +146,7 @@ function(vcpkg_acquire_msys PATH_TO_ROOT_OUT)
   endif()
 
   if(_am_PACKAGES)
-    message(STATUS "Acquiring MSYS Packages...")
+    message(STATUS "Acquiring MSYS Packages from ${TOOLPATH}...")
     string(REPLACE ";" " " _am_PACKAGES "${_am_PACKAGES}")
 
     set(_ENV_ORIGINAL $ENV{PATH})

